backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/appsearchmodal/internal/BpkAppSearchModalImpl.kt

```diff
@@ -35,6 +36,7 @@ internal fun BpkAppSearchModalImpl(
     inputHint: String,
     results: BpkAppSearchModalResult,
     onInputChanged: (String) -> Unit,
+    clearAction: BpkClearAction,
```

Comment:
rather than adding a new action here can we add a `Clear` type in the `BpkFieldStatus` with the same properties? that way we're not running into that situation where you can have clear AND validation/error, which is not possible and matches figma states as well

Reply:
Should we still add a callback for the clear event when we remove the clearAction?

Reply:
yes sorry that's what i meant with the same params - we still need the content description + callback

Reply:
sorry this wasn't clear - what i meant is that the `Clear` status should contain the properties here, rather than introducing a new `clearAction` property - otherwise we'll have conflicting sources for the clear action and can run into an invalid state (where status is clear, but the action is not provided)

Reply:
really sorry for the back and forth, BUT thinking this through again and looking at this in place it does make sense to have a separate clear action, rather than having a `BpkFieldStatus` of clear. few reasons for this:
the clear action is _only_ valid for the text field, not any other field set types
the logic to show/hide the clear action depending on content should be within text field to simplify the API (as you wrote it)
this also matches iOS API

one thing to change from the previous implementation, though: if we DO have a valid/error status this should take precedence over the clear action being available or not (ie if the status is valid/error we show that, rather than the clear action)

Reply:
Yes, it would make more sense.